### DISCUSSION TITLE
Bug or by design, using "Server relative path"

### DISCUSSION THREAD

**Discussion by TheRealStriff** [Barbarur/NovaPoint#14] [General]
Using remove file versions I targeted a folder in a document library.

Using the below details:  (edited company details)

Versions to Delete: Keep few versions
Number of versions to Keep: 101 / Delete
What to do when solution finishes: Remove account as Admin

Sites filter: Single Site
Site URL: https:/XXXX.sharepoint.com/sites/Lake
Library filter: Single library
Title: Secure Records
File filter: Specific location
Server relative path: /sites/Lake/Shared Documents/Technical Services/Civil

(Shared Documents library is labelled Secure Records)

Running the solution, it enumerated over 751000 items.
Which is the whole library.
The path I selected has only 12 files which the in the solutions folder> RemoveFileVersionsAuto shows.

So, it's enumerating the whole library when only targeting a section.

**Comment by Barbarur** [comment]
Hi @TheRealStriff ,

At this moment that's the design behavior. It needs to collect all items from the list, then it will process **only** the matching items.

Getting items by location crash if result is above 5k items due the viewthreshold, so at this moment I went safe and got them all, then filter.

I already have a draft code designed for trying getting only files in target location, then if get error because viewthreshold collect them all.

**Comment by Barbarur** [comment]
Hi @TheRealStriff ,

New version 0.7.0 implements retrieval of only the files under the target folder using the relative server URL up to 5,000 items. If a folder contains more than 5,000 items it hits view threshold and all files will be collected and filtered afterwards.

Please give it a try using the solution to delete versions.
